x-pack/legacy/plugins/siem/public/components/header_page_new/index.tsx

```diff
@@ -0,0 +1,220 @@
+import { LinkIcon, LinkIconProps } from '../link_icon';
+import { Subtitle, SubtitleProps } from '../subtitle';
+import * as i18n from './translations';
+
+interface HeaderProps {
+  border?: boolean;
+  isLoading?: boolean;
+}
+
+const Header = styled.header.attrs({
+  className: 'siemHeaderPage',
+})<HeaderProps>`
+  ${({ border, theme }) => css`
+    margin-bottom: ${theme.eui.euiSizeL};
+
+    ${border &&
+      css`
+        border-bottom: ${theme.eui.euiBorderThin};
+        padding-bottom: ${theme.eui.paddingSizes.l};
+        .euiProgress {
+          top: ${theme.eui.paddingSizes.l};
+        }
+      `}
+  `}
+`;
+Header.displayName = 'Header';
+
+const FlexItem = styled(EuiFlexItem)`
+  display: block;
+`;
+FlexItem.displayName = 'FlexItem';
+
+const LinkBack = styled.div.attrs({
+  className: 'siemHeaderPage__linkBack',
+})`
+  ${({ theme }) => css`
+    font-size: ${theme.eui.euiFontSizeXS};
+    line-height: ${theme.eui.euiLineHeight};
+    margin-bottom: ${theme.eui.euiSizeS};
+  `}
+`;
+LinkBack.displayName = 'LinkBack';
+
+const Badge = styled(EuiBadge)`
+  letter-spacing: 0;
+`;
+Badge.displayName = 'Badge';
+
+const StyledEuiBetaBadge = styled(EuiBetaBadge)`
+  vertical-align: middle;
+`;
+
+StyledEuiBetaBadge.displayName = 'StyledEuiBetaBadge';
+
+const StyledEuiButtonIcon = styled(EuiButtonIcon)`
+  ${({ theme }) => css`
+    margin-left: ${theme.eui.euiSize};
+  `}
+`;
+
+StyledEuiButtonIcon.displayName = 'StyledEuiButtonIcon';
+
+interface BackOptions {
+  href: LinkIconProps['href'];
+  text: LinkIconProps['children'];
+}
+
+interface BadgeOptions {
+  beta?: boolean;
+  text: string;
+  tooltip?: string;
+}
+
+interface DraggableArguments {
+  field: string;
+  value: string;
+}
+interface IconAction {
+  'aria-label': string;
+  iconType: string;
+  onChange: (a: string) => void;
+  onClick: (b: boolean) => void;
+  onSubmit: () => void;
+}
+
+export interface HeaderPageProps extends HeaderProps {
+  backOptions?: BackOptions;
+  badgeOptions?: BadgeOptions;
+  children?: React.ReactNode;
+  draggableArguments?: DraggableArguments;
+  isEditTitle?: boolean;
+  iconAction?: IconAction;
+  subtitle2?: SubtitleProps['items'];
+  subtitle?: SubtitleProps['items'];
+  title: string | React.ReactNode;
+}
+
+const HeaderPageComponent: React.FC<HeaderPageProps> = ({
+  backOptions,
+  badgeOptions,
+  border,
+  children,
+  draggableArguments,
+  isEditTitle,
+  iconAction,
+  isLoading,
+  subtitle,
+  subtitle2,
+  title,
+  ...rest
+}) => (
+  <Header border={border} {...rest}>
+    <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
+      <FlexItem grow={false}>
+        {backOptions && (
+          <LinkBack>
+            <LinkIcon href={backOptions.href} iconType="arrowLeft">
+              {backOptions.text}
+            </LinkIcon>
+          </LinkBack>
+        )}
+
+        {isEditTitle && iconAction ? (
+          <EuiFlexGroup alignItems="center" gutterSize="m" justifyContent="spaceBetween">
```

Comment:
We will make sense here to pass a props node like `editTitleNode` to be a React.Node or null. And if it is not null, we show the pencil close to the title and this component manages the state editTitleNode when the user clicks on the pencil. This will also allow us to manage different validation on our own without this component getting big.

Reply:
and if we do like that, it is just adding one prop to the old header_page component